### PR TITLE
Python API: fix ee.data.create_assets() when recursively creating parents

### PR DESCRIPTION
**Pull request by jbardo**
When calling ee.data.create_assets() with mk_parents=True parents, for example via the cli:

`earthengine create folder --parents projects/<project_id>/assets/a/b/c`

we systematically get the following error:

`TypeError: Parameter "name" value "projects/<project_id>/assets" does not match the pattern "^projects/[^/]+/assets/.*$"`

It seems the code still refers to an outdated organization `user/project` instead of `projects/<project_id>/assets`.

### CODE REVIEW

**Comment by bhagya0yashree** [comment]
import os

# Replace with your Google Earth Engine project ID
PROJECT_ID = "project id"

# Replace with the path to your folder containing .tif files
FOLDER_PATH = "path to the folder"

# Function to upload a single image
def upload_image(image_path, asset_id):
  """
  Uploads a .tif image to the specified GEE asset ID.
  """
  from earthengine import ee

  # Initialize Earth Engine
  ee.Initialize(authenticate=True)

  # Upload the image
  task = ee.batch.Upload.image(image_path, asset_id=asset_id, project=PROJECT_ID)
  task.start()
  print(f'Uploading {image_path} to {asset_id}...')
  task.await()
  print('Upload complete!')

# Loop through files in the folder and upload them
for filename in os.listdir(FOLDER_PATH):
  if filename.endswith('.tif'):
    # Construct asset ID with filename (replace with desired structure)
    asset_id = f'users/{os.path.splitext(filename)[0]}'
    image_path = os.path.join(FOLDER_PATH, filename)
    upload_image(image_path, asset_id) and this is the error message i'm getting  :   task.await()
         ^^^^^
SyntaxError: invalid syntax
 So basically I was trying upload image folder from my desktop to my GEE account using this python script but somehow I'm getting error messages and I'm not able to fix it. can @jbardo help me with this?

**Review by Tinkie22 — APPROVED** [review]

**Comment by dchirst** [comment]
Is there any progress on this? We face the same issue.
